pandas/core/indexing.py

```diff
@@ -1814,7 +1814,8 @@ def check_bool_indexer(ax, key):
         result = result.reindex(ax)
         mask = isnull(result._values)
         if mask.any():
-            raise IndexingError('Unalignable boolean Series key provided')
+            raise IndexingError('Unalignable labels in boolean Series index '
+                                '{}'.format(key.index))
```

Comment:
I think the suggestion of @jreback what to include `type(key.index)` instead of `key.index`. But anyway, I personally think that this full index will obfuscate the error message a bit, as the index can be typically much longer. 
Other proposal to make the message more clear:

```
Unalignable boolean Series provided as indexer (index of the boolean Series and of the indexed object do not match)
```

What do you think of that?